src/kaltura-player.js

```diff
@@ -648,6 +648,9 @@ class KalturaPlayer extends FakeEventTarget {
       this._eventManager.listen(this, AdEventType.AD_BREAK_END, () => this._attachMediaSource());
       this._eventManager.listen(this, AdEventType.AD_ERROR, () => this._attachMediaSource());
     }
+    this._eventManager.listen(this, 'floatingplayerstatechanged', e => {
```

Comment:
the fact that kaltura player has to know about plugin breaks isolation.
Better apply something more generic as API and let any plugin set this, so this is also future ready to other use cases

Reply:
Not relevant anymore after the code refactoring